src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -107,22 +110,32 @@ private Completable poll(String url) {
                         this.active = false;
                         this.closeError = "Unexpected response code " + response.getStatusCode() + ".";
                     } else {
-                        if (response.getContent() != null) {
+                        if (response.getContent() != null && response.getContent().hasRemaining()) {
                             logger.debug("Message received.");
-                            onReceiveThread.submit(() -> this.onReceive(response.getContent()));
+                            try {
+                                onReceiveThread.submit(() -> this.onReceive(response.getContent()));
+                            // it's possible for stop to be called while a request is in progress, if stop throws it wont wait for
+                            // an in-progress poll to complete and will shutdown the thread. We should ignore the exception so we don't
+                            // get an unhandled RX error
+                            } catch(Exception e) {}
                         } else {
                             logger.debug("Poll timed out, reissuing.");
                         }
                     }
-                    return poll(url);
+                    receiveLoopSubject.onNext(url);
```

Comment:
Does BehaviorSubject prevent the subscribers from running inline in this case?

Reply:
No, `onNext` still runs inline, the stackoverflow was from the error handler. Before we were returning new completables per poll which would extend the overall subscribe from `poll(url).subscribeWith(receiveLoop);`. But now we only subscribe to the `BehaviorSubject.onError(...)` so there wont be a continuous `onError` subscription from new completables.

**Before:**
![image](https://user-images.githubusercontent.com/7574801/122446198-3dbf8c80-cf57-11eb-99ae-83cdbb4eab0a.png)

**After:**
![image](https://user-images.githubusercontent.com/7574801/122446244-4ca63f00-cf57-11eb-8bed-092d9af21d1a.png)

Reply:
I think we should dispatch this then. While it seems unlikely that polls would always complete inline, it's doesn't seem impossible.

Reply:
I think in this case it will always be dispatched because of OkHttp, but I will look into doing the dispatching ourselves